src/thunderbots/software/ai/world/robot.cpp

```diff
@@ -92,12 +102,13 @@ Point Robot::estimatePositionAtFutureTime(const Duration &duration_in_future) co
     // real-world behavior. Position prediction should be improved as outlined in
     // https://github.com/UBC-Thunderbots/Software/issues/50
     double seconds_in_future = duration_in_future.getSeconds();
-    return position_ + velocity_.norm(velocity_.len() * seconds_in_future);
+    return positions_.back() +
+           velocities_.back().norm(velocities_.back().len() * seconds_in_future);
```

Comment:
instead of directly accessing the queues like this, we should be using the getter methods that we've already implemented. This makes it much easier to make changes in the future, and is less prone to mistakes. This goes for everywhere we're doing this (sans the getter functions themselves).

Reply:
Fixed